libraries/AP_Landing/AP_Landing_Slope.cpp

```diff
@@ -219,7 +221,7 @@ void AP_Landing::type_slope_adjust_landing_slope_for_rangefinder_bump(AP_Vehicle
         if (new_slope_deg - initial_slope_deg > slope_recalc_steep_threshold_to_abort) {
             gcs().send_text(MAV_SEVERITY_INFO, "Landing slope too steep, aborting (%.0fm %.1fdeg)",
                                              (double)rangefinder_state.correction, (double)(new_slope_deg - initial_slope_deg));
-            alt_offset = rangefinder_state.correction;
+            AP::baro().set_baro_drift_altitude( -1.0 * rangefinder_state.correction);
```

Comment:
this suffers from a few problems:
 - it only works for baro height, not GPS height
 - it assumes a landing discrepancy can only come from baro, when it could also be a terrain data error
 - it doesn't zero when you disarm
 - if user fetches parameters they get it on the GCS, and may load this offset later (general problem with using parameters for in flight corrections)
 - it doesn't handle a 2nd landing being at a different location

Reply:
I've updated to fix the existing correction instead